### DISCUSSION TITLE
稍后播放自动恢复上次外挂的音源

### DISCUSSION THREAD

**Discussion by YSYHZ** [hooke007/mpv_PlayKit#355] [mpv 通用]
这个需求的起因是b站的dash流是音视频分流的，使用Play-With-MPV这个脚本将b站视频推流到mpv后音源是外挂在视频流下面的
如下图所示：
![image](https://github.com/hooke007/MPV_lazy/assets/108722788/f766afd5-a28c-4826-8468-f3b586dc6177)
然后通过SimpleHistory.lua这个脚本打开历史播放的视频外挂音源消失，所以需要自动恢复。

看了[自动恢复上次外挂的字幕](https://github.com/hooke007/MPV_lazy/discussions/47)和[多重高级字幕并载](https://github.com/hooke007/MPV_lazy/discussions/61)后在mpv.conf中加入`save-position-on-quit=yes`
`watch_later_options`将会全部启用，在关闭视频后自动保存vf和af滤镜，成功完成了稍后播放自动恢复上次外挂的字幕。

目前的想法是有没有像subtitles这样外挂字幕的外挂音源af滤镜

**Comment by hooke007** [comment]
合并多轨音频的功能在脚本input_plus里有

**Reply by hooke007** [reply]
对象从sid变成了aid，一样内部无实现。
有需求参考第三方脚本自己做吧 https://github.com/zenwarr/mpv-config/blob/master/scripts/restore-subtitles.lua

**Reply by YSYHZ** [reply]
试了一下，放弃了，我连restore-subtitles.lua这个原脚本都用不明白成功写出了saved-subs.json这个文件，也用vsc打开看过里面的sub1也确实记录了我外挂的字幕，但是我看了一下saved-subs.json里并没有记录外挂的字幕的本地路径，再次打开文件字幕未成功恢复
